lib/fs/readlink/readlinknul.h

```diff
@@ -22,12 +22,12 @@
 
 
 ATTR_STRING(1)
-inline int readlinknul(const char *restrict link, char *restrict buf,
+inline ssize_t readlinknul(const char *restrict link, char *restrict buf,
     size_t size);
 
 
 // Similar to readlink(2), but terminate the string.
-inline int
+inline ssize_t
```

Comment:
Should you then return r instead of len?

Reply:
Hmmm, I guess you're right.

I'm not convinced by the name of the variable.  What do you think of the following options?

-  `r`
-  `slen`

Reply:
Maybe slen and ulen, for making it more obvious what the s and the u are?

Reply:
Done.  Thanks!